src/python/pants/engine/process_test.py

```diff
@@ -36,253 +25,106 @@ def rule_runner() -> RuleRunner:
+                            b'#!/bin/bash -eu\necho "Hello"\n',
+                            is_executable=is_executable,
+                        )
+                    ]
                 )
             ],
         )
-
-        req = Process(
-            argv=("/bin/cat", "a.txt", "b.txt"),
-            input_digest=digest,
-            description="cat the contents of this file",
-        )
-
-        result = self.request(ProcessResult, [req])
-        self.assertEqual(result.stdout, b"hellogoodbye")
-
-    def test_file_in_directory_creation(self):
-        path = "somedir/filename"
-        content = b"file contents"
-
-        digest = self.request(Digest, [CreateDigest([FileContent(path=path, content=content)])])
-        req = Process(
-            argv=("/bin/cat", "somedir/filename"),
-            input_digest=digest,
-            description="Cat a file in a directory to make sure that doesn't break",
-        )
-
-        result = self.request(ProcessResult, [req])
-        self.assertEqual(result.stdout, content)
-
-    def test_not_executable(self):
-        file_name = "echo.sh"
-        file_contents = b'#!/bin/bash -eu\necho "Hello"\n'
-
-        digest = self.request(
-            Digest, [CreateDigest([FileContent(path=file_name, content=file_contents)])]
-        )
-        req = Process(
-            argv=("./echo.sh",),
-            input_digest=digest,
-            description="cat the contents of this file",
-        )
-
-        with pytest.raises(ExecutionError) as exc:
-            self.request(ProcessResult, [req])
-        assert "Permission" in str(exc.value)
-
-    def test_executable(self):
-        file_name = "echo.sh"
-        file_contents = b'#!/bin/bash -eu\necho "Hello"\n'
-
-        digest = self.request(
-            Digest,
-            [
-                CreateDigest(
-                    [FileContent(path=file_name, content=file_contents, is_executable=True)]
-                )
-            ],
-        )
-        req = Process(
+        process = Process(
             argv=("./echo.sh",),
             input_digest=digest,
             description="cat the contents of this file",
         )
+        return rule_runner.request(ProcessResult, [process])
 
-        result = self.request(ProcessResult, [req])
-        self.assertEqual(result.stdout, b"Hello\n")
+    assert run_process(is_executable=True).stdout == b"Hello\n"
 
+    with pytest.raises(ExecutionError) as exc:
+        run_process(is_executable=False)
+    assert "Permission" in str(exc.value)
 
-class ProcessTest(TestBase):
-    @classmethod
-    def rules(cls):
-        return (
-            *super().rules(),
-            *cat_stdout_rules(),
-            QueryRule(ProcessResult, (Process,)),
-            QueryRule(FallibleProcessResult, (Process,)),
-        )
 
-    def test_env(self):
-        req = Process(argv=("foo",), description="Some process", env={"VAR": "VAL"})
-        assert dict(req.env) == {"VAR": "VAL"}
```

Comment:
This original test wasn't really testing much..rewritten to be more comprehensive.